src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -273,7 +281,13 @@ protected override void ContextUnloaded(ExecutionContext context)
                     {
                         ExecutionContextState contextState = CurrentContext.GetState<ExecutionContextState>();
                         contextState.NotificationCount += state.NotificationCount;
-                        if (state.PushNullWhenReturn) Push(StackItem.Null);
+                        if (state.IsDynamicCall)
+                        {
+                            if (context.EvaluationStack.Count == 0)
+                                Push(StackItem.Null);
+                            else if (context.EvaluationStack.Count > 1)
```

Comment:
Isn't this duplicating https://github.com/neo-project/neo-vm/blob/f48de7e595ade7c72be6ccd5cce0ab99708f0afd/src/Neo.VM/ExecutionEngine.cs#L427 (rvcount normally is 0 or 1 in this case: https://github.com/neo-project/neo/blob/8e68c3fabf8b7cad3bd27e0c556cbeda17c2b123/src/Neo/SmartContract/ApplicationEngine.cs#L331)? Can this ever happen without #2756?

Reply:
```c#
rvcount: method.ReturnType == ContractParameterType.Void ? 0 : 1,
```

This is in `LoadContract()`. For a dynamic script, we have no way of knowing in advance how many values it will return. So set `rvcount` to `-1`.

https://github.com/neo-project/neo-vm/blob/master/src/Neo.VM/ExecutionEngine.cs#L1580

Reply:
Yeah. I mean, this change probably can be a part of #2756, it doesn't seem to be useful without it.

Reply:
We can use it as a double check.